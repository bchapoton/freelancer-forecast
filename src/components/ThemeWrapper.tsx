import React, { ReactNode } from 'react';
import { createTheme, ThemeProvider } from '@mui/material';

export type ThemeWrapperProps = {
    children: ReactNode;
};

function ThemeWrapper({ children }: ThemeWrapperProps) {
    const theme = createTheme({
        typography: {
            fontSize: 13,
        },
        components: {
            MuiTableRow: {
                styleOverrides: {
                    head: {
                        color: '#495057',
                        backgroundColor: '#eff2f7',
                        fontWeight: 600,
                        borderBottomColor: '#d7dade',
                        borderBottomWidth: '2px',
                        borderBottomStyle: 'solid',
                    },
                    hover: {
                        backgroundColor: '#495057',
                    },
                    root: {
                        '&:last-child td, &:last-child th': { border: 0 },
                    },
                },
            },
            MuiTableCell: {
                styleOverrides: {
                    head: {
                        fontSize: '13px',
                        fontWeight: 600,
                        color: '#495057',
                    },
                    body: {
                        color: '#495057',
                        fontSize: '13px',
                    },
                },
            },
            MuiTableContainer: {
                styleOverrides: {
                    root: {
                        borderRadius: 5,
                    },
                },
            },
        },
    });
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export default ThemeWrapper;
