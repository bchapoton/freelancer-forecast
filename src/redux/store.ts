import { configureStore } from '@reduxjs/toolkit';
import parametersReducer from './slices/ParametersSlice';
import summariesReducer from './slices/SummariesSlice';
import uiReducer from './slices/UISlice';
import { StorePreloadedState } from '../helpers/TestHelper';

export const store = storeConfigurator();

export function storeConfigurator(preloadedState?: StorePreloadedState) {
    return configureStore({
        reducer: {
            parameters: parametersReducer,
            summaries: summariesReducer,
            ui: uiReducer,
        },
        preloadedState: preloadedState,
    });
}

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
