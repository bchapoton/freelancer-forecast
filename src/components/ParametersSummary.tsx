import React, { Fragment, ReactNode } from 'react';
import { Box, Card, CardContent, Typography } from '@mui/material';
import { useAppSelector } from '../redux/hooks';
import { Saving, selectParameters, SocialContribution } from '../redux/slices/ParametersSlice';
import Euro from './Euro';
import Percentage from './Percentage';
import calendarService from '../services/CalendarService';
import { SxProps } from '@mui/system';
import { Theme } from '@mui/material/styles';
import DayFormatter from './DayFormatter';

function ParametersSummary() {
    const parameters = useAppSelector(selectParameters);

    return (
        <Fragment>
            <Typography gutterBottom variant="h5" component="div">
                Paramètres
            </Typography>
            <BoxedParametersContainer>
                <CardParameters
                    title="Informations du freelance"
                    parameters={[
                        { label: 'TJM', value: <Euro cents={false}>{parameters.averageDailyRate}</Euro> },
                        { label: 'Temps de travail', value: <Percentage>{parameters.partTime.percentage}</Percentage> },
                        {
                            label: 'Congés',
                            value: calendarService.formatDayOffContext(parameters.dayOffContext),
                        },
                        { label: 'TVA', value: <Percentage>{parameters.vatRate}</Percentage> },
                        { label: 'Abattement fiscal', value: <Percentage>{parameters.taxAllowance}</Percentage> },
                        { label: "Epargne d'entreprise", value: <ParameterSaving saving={parameters.saving} /> },
                        {
                            label: 'Jours non ouvrable',
                            value: <DayFormatter>{parameters.nonWorkingDays}</DayFormatter>,
                        },
                    ]}
                />
                <CardParameters
                    title="Cotisations sociales"
                    parameters={parameters.socialContribution.map((socialContribution: SocialContribution) => {
                        return {
                            label: socialContribution.label,
                            value: <Percentage>{socialContribution.rate}</Percentage>,
                        };
                    })}
                />
                <CardParameters
                    title="Contexte familial fiscal"
                    parameters={[
                        { label: 'Nombre de parts', value: parameters.taxationFamilyContext.incomeSplittingParts },
                        {
                            label: 'Revenus du foyer',
                            value: <Euro>{parameters.taxationFamilyContext.taxableHouseholdRevenues}</Euro>,
                        },
                        {
                            label: 'Abattement fiscal',
                            value: <Percentage>{parameters.taxationFamilyContext.taxAllowance}</Percentage>,
                        },
                    ]}
                />
            </BoxedParametersContainer>
        </Fragment>
    );
}

export default ParametersSummary;

type SavingProps = {
    saving: Saving;
};

function ParameterSaving({ saving }: SavingProps) {
    if (saving.mode === 'percentage') return <Percentage>{saving.value}</Percentage>;
    else return <Euro>{saving.value}</Euro>;
}

type CardParametersProps = {
    title: string;
    parameters: {
        label: string;
        value: ReactNode;
    }[];
};

function CardParameters({ title, parameters }: CardParametersProps) {
    return (
        <Card>
            <CardContent sx={{ p: '10px' }}>
                <Typography
                    gutterBottom
                    variant="h5"
                    component="div"
                    sx={{ fontSize: '1.2rem', fontWeight: 600, color: '#33383d' }}
                >
                    {title}
                </Typography>
                {parameters.map((parameter, index) => (
                    <BoxedParameters key={'card-parameter' + index} title={parameter.label} value={parameter.value} />
                ))}
            </CardContent>
        </Card>
    );
}

export type BoxedParametersProps = {
    title: ReactNode;
    value: ReactNode;
    noWrap?: boolean;
    highlighted?: boolean;
};

export function BoxedParameters({ title, value, noWrap = true, highlighted = false }: BoxedParametersProps) {
    let titleTypographySxValue: SxProps<Theme> = {
        fontWeight: 600,
        fontSize: 13,
        color: '#495057',
        width: '50%',
    };

    if (noWrap) {
        titleTypographySxValue = { ...titleTypographySxValue, maxWidth: '250px', minWidth: '150px' };
    }

    let boxSxValue: SxProps<Theme> = {
        display: 'flex',
    };
    if (highlighted) {
        boxSxValue = {
            ...boxSxValue,
            backgroundColor: '#f8f9fa',
            borderTopStyle: 'solid',
            borderTopWidth: 1,
            borderTopColor: '#d7dade',
        };
    }

    return (
        <Box sx={boxSxValue}>
            <Typography noWrap={noWrap} variant="subtitle2" sx={titleTypographySxValue} component={'div'}>
                {title}
            </Typography>
            <Typography variant="body2" component={'div'} sx={{ width: '50%', textAlign: 'right' }}>
                {value}
            </Typography>
        </Box>
    );
}

export type BoxedParametersContainerProps = {
    children: ReactNode;
};

export function BoxedParametersContainer({ children }: BoxedParametersContainerProps) {
    return <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-evenly' }}>{children}</Box>;
}
