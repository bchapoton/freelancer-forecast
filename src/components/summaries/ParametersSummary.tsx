import React, { ReactNode, useEffect, useState } from 'react';
import { Box, Card, CardContent, Collapse, Grid, IconButton, Typography, useMediaQuery } from '@mui/material';
import { useAppSelector } from '../../redux/hooks';
import { Saving, selectParameters, SocialContribution } from '../../redux/slices/ParametersSlice';
import Euro from '../formatters/Euro';
import Percentage from '../formatters/Percentage';
import calendarService from '../../services/CalendarService';
import { SxProps } from '@mui/system';
import { Theme } from '@mui/material/styles';
import DayFormatter from '../formatters/DayFormatter';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ExpandLessIcon from '@mui/icons-material/ExpandLess';

function ParametersSummary() {
    const parameters = useAppSelector(selectParameters);
    const [collapsed, setCollapsed] = useState<boolean>(false);
    const mediaQuery: boolean = useMediaQuery((theme: Theme) => theme.breakpoints.down('md'));

    useEffect(() => {
        if (mediaQuery) {
            setCollapsed(true);
        }
    }, [mediaQuery]);

    return (
        <Box sx={{ mt: '10px', mb: '10px' }}>
            <Box
                sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', mt: '10px', mn: '10px' }}
            >
                <Typography gutterBottom variant="h5" component="div">
                    Paramètres
                </Typography>
                <IconButton onClick={() => setCollapsed((collapsed) => !collapsed)}>
                    {collapsed ? <ExpandMoreIcon /> : <ExpandLessIcon />}
                </IconButton>
            </Box>
            <Collapse in={!collapsed} timeout="auto" unmountOnExit component={Box}>
                <Grid container spacing={3}>
                    <ParameterGrid>
                        <CardParameters
                            items={[
                                {
                                    title: 'Informations du freelance',
                                    parameters: [
                                        {
                                            label: 'TJM',
                                            value: <Euro cents={false}>{parameters.averageDailyRate}</Euro>,
                                        },
                                        {
                                            label: 'Temps de travail',
                                            value: <Percentage>{parameters.partTime.percentage}</Percentage>,
                                        },
                                        {
                                            label: 'Congés',
                                            value: calendarService.formatDayOffContext(parameters.dayOffContext),
                                        },

                                        {
                                            label: "Epargne d'entreprise",
                                            value: <ParameterSaving saving={parameters.saving} />,
                                        },
                                        {
                                            label: 'Jours non ouvrables',
                                            value: <DayFormatter>{parameters.nonWorkingDays}</DayFormatter>,
                                        },
                                    ],
                                },
                            ]}
                        />
                    </ParameterGrid>
                    <ParameterGrid>
                        <CardParameters
                            items={[
                                {
                                    title: 'Cotisations sociales',
                                    parameters: parameters.socialContribution.map(
                                        (socialContribution: SocialContribution) => {
                                            return {
                                                label: socialContribution.label,
                                                value: <Percentage>{socialContribution.rate}</Percentage>,
                                            };
                                        },
                                    ),
                                },
                                {
                                    title: 'Fiscalité entreprise',
                                    parameters: [
                                        { label: 'TVA', value: <Percentage>{parameters.vatRate}</Percentage> },
                                        {
                                            label: 'Abattement forfaitaire',
                                            value: <Percentage>{parameters.taxAllowance}</Percentage>,
                                        },
                                    ],
                                },
                            ]}
                        />
                    </ParameterGrid>
                    <ParameterGrid>
                        <CardParameters
                            items={[
                                {
                                    title: 'Fiscalité personnelle',
                                    parameters: [
                                        {
                                            label: 'Nombre de parts',
                                            value: parameters.taxationFamilyContext.incomeSplittingParts,
                                        },
                                        {
                                            label: 'Revenus du foyer',
                                            value: (
                                                <Euro>{parameters.taxationFamilyContext.taxableHouseholdRevenues}</Euro>
                                            ),
                                        },
                                        {
                                            label: 'Abattement forfaitaire',
                                            value: (
                                                <Percentage>{parameters.taxationFamilyContext.taxAllowance}</Percentage>
                                            ),
                                        },
                                    ],
                                },
                            ]}
                        />
                    </ParameterGrid>
                </Grid>
            </Collapse>
        </Box>
    );
}

export default ParametersSummary;

type SavingProps = {
    saving: Saving;
};

function ParameterSaving({ saving }: SavingProps) {
    if (saving.mode === 'percentage') return <Percentage>{saving.value}</Percentage>;
    else return <Euro>{saving.value}</Euro>;
}

type CardParametersProps = {
    items: CardParametersItemProps[];
};

function CardParameters({ items }: CardParametersProps) {
    return (
        <CardContent sx={{ p: '10px' }}>
            {items.map((itemProps: CardParametersItemProps, index) => (
                <CardParametersItem key={'item-' + index} {...itemProps} />
            ))}
        </CardContent>
    );
}

type CardParametersItemProps = {
    title: string;
    parameters: {
        label: string;
        value: ReactNode;
    }[];
};

function CardParametersItem({ title, parameters }: CardParametersItemProps) {
    return (
        <Box>
            <Typography
                gutterBottom
                variant="h5"
                component="div"
                sx={{ fontSize: '1.2rem', fontWeight: 600, color: '#33383d' }}
            >
                {title}
            </Typography>
            {parameters.map((parameter, index) => (
                <BoxedParameters key={'card-parameter' + index} title={parameter.label} value={parameter.value} />
            ))}
        </Box>
    );
}

export type BoxedParametersProps = {
    title: ReactNode;
    value: ReactNode;
    noWrap?: boolean;
    highlighted?: boolean;
};

export function BoxedParameters({ title, value, noWrap = true, highlighted = false }: BoxedParametersProps) {
    let titleTypographySxValue: SxProps<Theme> = {
        fontWeight: 600,
        fontSize: 13,
        color: '#495057',
        width: '50%',
    };

    if (noWrap) {
        titleTypographySxValue = { ...titleTypographySxValue, maxWidth: '50%' };
    }

    let boxSxValue: SxProps<Theme> = {
        display: 'flex',
    };
    if (highlighted) {
        boxSxValue = {
            ...boxSxValue,
            backgroundColor: '#f8f9fa',
            borderTopStyle: 'solid',
            borderTopWidth: 1,
            borderTopColor: '#d7dade',
        };
    }

    return (
        <Box sx={boxSxValue}>
            <Typography noWrap={noWrap} variant="subtitle2" sx={titleTypographySxValue} component={'div'}>
                {title}
            </Typography>
            <Typography variant="body2" component={'div'} sx={{ width: '50%', textAlign: 'right' }}>
                {value}
            </Typography>
        </Box>
    );
}

export type ParameterGridProps = {
    children: ReactNode;
};

export function ParameterGrid({ children }: ParameterGridProps) {
    return (
        <Grid item xs={12} md={4} sx={{ padding: '5px' }}>
            <Card>{children}</Card>
        </Grid>
    );
}
