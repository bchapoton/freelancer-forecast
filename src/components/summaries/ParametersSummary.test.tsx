import ParametersSummary from './ParametersSummary';
import React from 'react';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store';
import { Provider } from 'react-redux';
import ThemeWrapper from '../ThemeWrapper';

it('Parameters summary renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <ThemeWrapper>
                    <ParametersSummary />
                </ThemeWrapper>
            </Provider>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
