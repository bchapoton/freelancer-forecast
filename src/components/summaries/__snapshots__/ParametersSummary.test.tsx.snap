// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parameters summary renders correctly 1`] = `
<div
  className="MuiBox-root css-1xo3mso"
>
  <div
    className="MuiBox-root css-ct6m7r"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1pzmhef-MuiTypography-root"
    >
      Paramètres
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m1cnyu-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xkvp8t-MuiSvgIcon-root"
        data-testid="ExpandLessIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m12 8-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
        />
      </svg>
    </button>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered MuiBox-root css-uhoni6-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-gn09uc-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-fpqqwi-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-0"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-a7rkoh-MuiTypography-root"
                  >
                    Informations du freelance
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      TJM
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        650 €
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Temps de travail
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        100
                        %
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Congés
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      5 semaines
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Epargne d'entreprise
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        20
                        %
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Jours non ouvrable
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        sam., dim.
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-gn09uc-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-fpqqwi-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-0"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-a7rkoh-MuiTypography-root"
                  >
                    Cotisations sociales
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Prestations de services (bnc et bic) et vente de marchandises (bic)
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        22.2
                        %
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Formation prof.liberale obligatoire
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        0.2
                        %
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiBox-root css-0"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-a7rkoh-MuiTypography-root"
                  >
                    Fiscalité entreprise
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      TVA
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        20
                        %
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Abattement fiscal
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        34
                        %
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-gn09uc-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-fpqqwi-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-0"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-a7rkoh-MuiTypography-root"
                  >
                    Contexte fiscal personnel
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Nombre de parts
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      3
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Revenus du foyer
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        24 506,00 €
                      </span>
                    </div>
                  </div>
                  <div
                    className="MuiBox-root css-k008qs"
                  >
                    <div
                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-noWrap css-1pc1uhe-MuiTypography-root"
                    >
                      Abattement fiscal
                    </div>
                    <div
                      className="MuiTypography-root MuiTypography-body2 css-1umiv4o-MuiTypography-root"
                    >
                      <span>
                        10
                        %
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
